gateway-ha/src/main/java/io/trino/gateway/ha/handler/RoutingTargetHandler.java

```diff
@@ -81,8 +82,10 @@ public RoutingTargetResponse resolveRouting(HttpServletRequest request)
         RoutingTargetResponse routingTargetResponse = previousCluster.map(cluster -> {
             String routingGroup = queryId.map(routingManager::findRoutingGroupForQueryId)
                     .orElse("adhoc");
+            String externalUrl = queryId.map(routingManager::findExternalUrlForQueryId)
+                    .orElse(cluster);
             return new RoutingTargetResponse(
-                    new RoutingDestination(routingGroup, cluster, buildUriWithNewCluster(cluster, request)),
+                    new RoutingDestination(routingGroup, cluster, buildUriWithNewCluster(cluster, request), externalUrl),
```

Comment:
Adding external URL to RoutingDestination is meant only for logging at queryHistoryManger right?
And this is needed to show it at History page?

If that's the only use, then it may be more than needed to include it in every RoutingDestination, as this object is passed through multiple layers. Passing it through the entire routing stack, rather than fetching it only where needed, introduces extra coupling and data flow.

@vishalya Any thoughts on this?